general/echo/kmdf/exe/src/main.rs

```diff
@@ -0,0 +1,489 @@
+use once_cell::sync::Lazy;
+use std::os::windows::prelude::*;
+use std::thread;
+use std::{env, error::Error, ffi::OsString, sync::RwLock};
+use uuid::{uuid, Uuid};
+use windows_sys::Win32::Foundation::{CloseHandle, ERROR_IO_PENDING};
+use windows_sys::Win32::System::Threading::INFINITE;
+use windows_sys::Win32::System::IO::{
+    CreateIoCompletionPort, GetQueuedCompletionStatus, OVERLAPPED, OVERLAPPED_0,
+};
+use windows_sys::Win32::{
+    Foundation::{GetLastError, HANDLE, INVALID_HANDLE_VALUE},
+    Storage::FileSystem::*,
+};
+
+#[derive(Default, Debug)]
+struct Globals {
+    g_perform_async_io: bool,
+    g_limited_loops: bool,
+    g_async_io_loops_num: usize,
+    g_device_path: String,
+}
+
+static GLOBAL_DATA: Lazy<RwLock<Globals>> = Lazy::new(|| RwLock::new(Globals::default()));
```

Comment:
For the async path, a number of threads are started and need access to Global in parallel.  Rather than cloning global, RwLock let's them all hold a read lock as long as they are running, run in parallel and autodrop at the end.